front/src/applications/stdcmV2/views/StdcmViewV2.tsx

```diff
@@ -1,66 +1,189 @@
 import React, { useEffect, useState } from 'react';
 
 import cx from 'classnames';
+import { isEqual } from 'lodash';
 import { useTranslation } from 'react-i18next';
 import { useSelector } from 'react-redux';
 
 import useStdcm from 'applications/stdcm/hooks/useStdcm';
-import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useOsrdConfSelectors, useOsrdConfActions } from 'common/osrdContext';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import { useAppDispatch } from 'store';
 
 import StdcmConfig from '../components/StdcmConfig';
 import StdcmHeader from '../components/StdcmHeader';
 import StdcmResults from '../components/StdcmResults';
-import type { StdcmSimulationResult } from '../types';
+import type { StdcmSimulation, StdcmSimulationInputs } from '../types';
 
 const StdcmViewV2 = () => {
-  const { t } = useTranslation('stdcm');
-  const { launchStdcmRequest, cancelStdcmRequest, isPending, stdcmV2Results, pathProperties } =
-    useStdcm();
+  const {
+    launchStdcmRequest,
+    cancelStdcmRequest,
+    isPending,
+    isSuccessful,
+    isRejected,
+    isStdcmResultsEmpty,
+    stdcmV2Results,
+    pathProperties,
+  } = useStdcm(false);
   const { getScenarioID } = useOsrdConfSelectors();
   const scenarioID = useSelector(getScenarioID);
 
-  const [currentSimulationInputs, setCurrentSimulationInputs] = useState<
-    StdcmSimulationResult['input'] | undefined
-  >(undefined);
+  const [currentSimulationInputs, setCurrentSimulationInputs] = useState<StdcmSimulationInputs>();
```

Comment:
To stay closer to the store behavior, do you think we could initiate the state like that ?
This way, you can : 
- remove the "?" in the type definition of the state for pathSteps
- remove the "undefined" checks where you update the state
- remove the check "where to replaceAtIndex" (it will always be at the end of the array as it's always defined) when updating destination

Optional : we could

```suggestion
  const [currentSimulationInputs, setCurrentSimulationInputs] = useState<StdcmSimulationInputs>({
    pathSteps: [null, null],
  });
```

Reply:
Good point !

Reply:
Done